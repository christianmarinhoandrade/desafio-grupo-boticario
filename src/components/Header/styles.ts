import styled, { css } from 'styled-components'

import { Layout } from 'antd'
import { HeaderProps } from '.'

const HeaderAntd = Layout.Header

export const Header = styled(HeaderAntd)<HeaderProps>`
  display: flex;
  align-items: center;
  ${(props) => css`
    justify-content: ${props.justifycontent};
  `};
`
