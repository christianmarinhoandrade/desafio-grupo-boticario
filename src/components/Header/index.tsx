import * as S from './styles'
export type HeaderProps = {
  children: JSX.Element
  justifycontent?: 'space-around' | 'flex-end'
}
function Header({
  children,
  justifycontent = 'space-around',
  ...props
}: HeaderProps) {
  return (
    <S.Header justifycontent={justifycontent} {...props}>
      {children}
    </S.Header>
  )
}

export default Header
