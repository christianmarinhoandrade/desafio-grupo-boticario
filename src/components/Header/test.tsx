import { render, screen } from '@testing-library/react'

import Header from '.'

describe('<Header />', () => {
  it('should render header default', () => {
    const { container } = render(
      <Header data-testid={'header-id'}>
        <div />
      </Header>
    )
    expect(screen.getByTestId('header-id')).toBeInTheDocument()
    expect(screen.getByTestId('header-id')).toHaveStyle({
      'justify-content': 'space-around'
    })
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render header with justify-content', () => {
    render(
      <Header data-testid="header-id" justifycontent={'flex-end'}>
        <div />
      </Header>
    )
    expect(screen.getByTestId('header-id')).toHaveStyle({
      'justify-content': 'flex-end'
    })
  })
})
