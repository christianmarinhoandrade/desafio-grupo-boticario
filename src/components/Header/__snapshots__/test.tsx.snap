// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render header default 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

<header
  class="ant-layout-header c0"
  data-testid="header-id"
  justifycontent="space-around"
>
  <div />
</header>
`
